sentry-android-core/src/test/java/io/sentry/android/core/ActivityLifecycleIntegrationTest.kt

```diff
@@ -420,4 +421,60 @@ class ActivityLifecycleIntegrationTest {
 
         verify(fixture.hub).captureTransaction(any())
     }
+
+    @Test
+    fun `App start is Cold when savedInstanceState is null`() {
+        val sut = fixture.getSut(14)
+        fixture.options.tracesSampleRate = 1.0
+        sut.register(fixture.hub, fixture.options)
+
+        val activity = mock<Activity>()
+        sut.onActivityCreated(activity, null)
+
+        assertTrue(AppStartState.getInstance().coldStart)
+    }
+
+    @Test
+    fun `App start is Warm when savedInstanceState is not null`() {
+        val sut = fixture.getSut(14)
+        fixture.options.tracesSampleRate = 1.0
+        sut.register(fixture.hub, fixture.options)
+
+        val activity = mock<Activity>()
+        val bundle = Bundle()
+        sut.onActivityCreated(activity, bundle)
+
+        assertFalse(AppStartState.getInstance().coldStart)
+    }
+
+    @Test
+    fun `Do not overwrite App start type after set`() {
+        val sut = fixture.getSut(14)
+        fixture.options.tracesSampleRate = 1.0
+        sut.register(fixture.hub, fixture.options)
+
+        val activity = mock<Activity>()
+        val bundle = Bundle()
+        sut.onActivityCreated(activity, bundle)
+        sut.onActivityCreated(activity, null)
+
+        assertFalse(AppStartState.getInstance().coldStart)
+    }
+
+    @Test
+    fun `App start end time is set`() {
+        val sut = fixture.getSut(14)
+        fixture.options.tracesSampleRate = 1.0
+        sut.register(fixture.hub, fixture.options)
+
+        // set by SentryPerformanceProvider so forcing it here
+        AppStartState.getInstance().setAppStartTime(0, Date())
+
+        val activity = mock<Activity>()
+        sut.onActivityCreated(activity, null)
+        sut.onActivityResumed(activity)
+
+        // SystemClock.uptimeMillis() always returns 0, can't assert real values
```

Comment:
`m`: We could use a wrapper around `SystemClock.uptimeMillis()` to make it testable or do, for example, `AppStartState.getInstance().setAppStartTime(-1.5, Date())` to make it testable.

Reply:
yep, just wanted to make it simple now, similar provider is `CurrentDateProvider`

Reply:
its not possible due to `SentryPerformanceProvider` relies on class load time, which is before the ctor of the class, so I cant really mock `SystemClock.uptimeMillis()` for that use case.